server/db/014-addrespondedattosessions.sql

```diff
@@ -0,0 +1,21 @@
+DO $migration$
+    DECLARE migrationId INT;
+    DECLARE lastSuccessfulMigrationId INT;
+BEGIN
+    -- The migration ID of this file
+    migrationId := 14;
+
+    -- Get the migration ID of the last file to be successfully run
+    SELECT MAX(id) INTO lastSuccessfulMigrationId
+    FROM migrations;
+
+    -- Only execute this script if its migration ID is next after the last successful migration ID
+    IF migrationId - lastSuccessfulMigrationId = 1 THEN
+        ALTER TABLE sessions 
+        ADD COLUMN responded_at timestamptz;
+
```

Comment:
What do you think about us setting the `responded_at` value to the `updated_at` value for all existing rows in `sessions` (after first temporarily turning off the updated_at trigger for this table)? Or maybe only for the rows which aren't 'Started' or 'Waiting for reply'. We'd sort of be lying because it won't necessarily be the time when the Button press was first responded to, but in most cases, I think it would be close.

Reply:
I don't think this is necessary, since `responded_at` will need to accommodate null values anyhow in order to accommodate unresponded sessions. also, we may want to calculate something like average response time in the future, and backfilling `responded_at` in this way might introduce some outliers that could skew the result of calculations like this.

if I remember correctly, @trisapeace you were talking about the old sessions rendering in a misleading way if `responded_at` is null for sessions that actually got a response, but maybe the fronted can handle that by looking at the session state?

Reply:
actually, we could backfill `responded_at` for rows where state is 'Waiting for category', since those are the rows where the last update was the initial response. this assumes that no other batch operations or maintenance have affected the `updated_at` value.

Reply:
but I think I would still prefer to leave the old sessions with blank `responded_at` values for now - we can always backfill them later if it becomes a problem.

Reply:
Yes, I was concerned that for Sessions that obviously had been responded to (like they had an incident category assigned to them), would display "Alert response: No response" in the expanded part of the Historic Alert view like the bottom view in this screenshot.
![image (9)](https://user-images.githubusercontent.com/1490437/121966513-a9211880-cd23-11eb-83e7-9c2111046c16.png)

Yes, we could work in some logic in the Alert App to not do this, but I'm thinking that by the time the Alert App is live, all of the Sessions that would be returned will have a Response Time.

Reply:
makes sense.